build_tools/benchmarks/run_benchmarks_on_linux.py

```diff
@@ -66,20 +66,23 @@ def run_benchmark_case(self, benchmark_case: BenchmarkCase,
                          taskset=taskset,
                          run_flags=run_flags)
 
-  def __parse_flagfile(self, case_dir: str) -> List[str]:
-    with open(os.path.join(case_dir, MODEL_FLAGFILE_NAME)) as flagfile:
+  def __parse_flagfile(self, case_dir: pathlib.Path) -> List[str]:
+    with (case_dir / MODEL_FLAGFILE_NAME).open("r") as flagfile:
       return [line.strip() for line in flagfile.readlines()]
 
   def __run_benchmark(self, benchmark_case: BenchmarkCase,
-                      results_filename: str, taskset: str,
+                      results_filename: pathlib.Path, taskset: str,
                       run_flags: List[str]):
+    if self.config.normal_benchmark_tool_dir is None:
+      raise ValueError("normal_benchmark_tool_dir can't be None.")
+
     tool_name = benchmark_case.benchmark_tool_name
-    tool_path = os.path.join(self.config.normal_benchmark_tool_dir, tool_name)
+    tool_path = str(self.config.normal_benchmark_tool_dir / tool_name)
```

Comment:
You don't need to do the explicit string conversion. subprocess does it automatically

Reply:
Done.